bb8/src/internals.rs

```diff
@@ -49,7 +49,17 @@ where
 
     pub(crate) fn reap(&self) -> ApprovalIter {
         let mut locked = self.internals.lock();
-        locked.reap(&self.statics)
+        let (iter, max_idle_timeout_closed, max_lifetime_closed) = locked.reap(&self.statics);
+        drop(locked);
+
+        self.statistics
+            .record(StatsKind::ConnectionsMaxLifeTimeClosed, max_lifetime_closed);
+        self.statistics.record(
+            StatsKind::ConnectionsMaxIdleTimeoutClosed,
+            max_idle_timeout_closed,
+        );
```

Comment:
Let's wrap this in a separate method on `AtomicStatistics`, too, and keep all of the reaping-related changes in a separate commit or PR.